chain/src/main/java/com/iluwatar/chain/RequestHandler.java

```diff
@@ -44,12 +44,15 @@ public RequestHandler(RequestHandler next) {
    * Request handler
    */
   public void handleRequest(Request req) {
+    if (!req.isHandled()) {
+      handleHook(req);
+    }
```

Comment:
This causes `xxx handling request xxx` to be printed although they are not handling it

Reply:
right, it can be so, simplifyied even more. need to pull, than, request type as an input parameter into constructor